processing/src/main/java/io/druid/query/DataSourceUtil.java

```diff
@@ -28,4 +33,22 @@ public static String getMetricName(DataSource dataSource)
     final List<String> names = dataSource.getNames();
     return names.size() == 1 ? names.get(0) : names.toString();
   }
+
+  public static String getMetricName(Iterable<DataSourceWithSegmentSpec> dataSources)
+  {
+    return StreamSupport.stream(dataSources.spliterator(), false)
```

Comment:
Note that `Iterables.toString()` does this, however it adds spaces after commas.